rust/openvasd/src/storage/mod.rs

```diff
@@ -223,3 +223,11 @@ impl<T> Storage for T where
     T: ProgressGetter + ScanStorer + AppendFetchResult + NVTStorer + ScanIDClientMapper
 {
 }
+
+/// A storage type that can be created from a Config and a list of feeds.
+pub trait FromConfigAndFeeds: Storage + Sized {
+    fn from_config_and_feeds(
+        config: &Config,
+        feeds: Vec<FeedHash>,
+    ) -> Result<Self, Box<dyn std::error::Error + Send + Sync>>;
+}
```

Comment:
Not super happy about this name or even the existence of this trait, but I felt like it would be nice to move the internal construction logic of each individual storage type out of main, so that we can just call `from_config_and_feeds(...)?` in main and be done with it. This could also have been a `TryFrom<(&Config, Vec<FeedHash>)>` impl or a method on the `Storage` trait itself.